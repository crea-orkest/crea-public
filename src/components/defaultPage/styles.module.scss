@use 'sass:map';
@use '../../styles/tokens/spacing.scss' as spacings;
@use '../../styles/tokens/breakpoints.scss' as breakpoints;
@use '../../styles/tokens/layout.scss' as layout;

.article {
  width: 100%;
  padding: 0 map.get(spacings.$px, 24);

  @media screen and (min-width: breakpoints.$sm) {
    max-width: layout.$default-width;
    padding: 0 map.get(spacings.$px, 64);
    margin: 0 auto;
  }
}
