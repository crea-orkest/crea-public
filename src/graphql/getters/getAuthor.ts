import type { RequestInit } from 'next/dist/server/web/spec-extension/request'
import { authorFormatter } from '../formatters/authorFormatter'
import { client } from '../gqlClient'
import {
  GetAuthorDocument,
  type GetAuthorQuery,
  type GetAuthorQueryVariables,
} from '../generated/graphql'

export const getAuthor = async (
  { id }: GetAuthorQueryVariables,
  fetchOptions?: RequestInit
) => {
  try {
    const { data, error } = await client.query<
      GetAuthorQuery,
      GetAuthorQueryVariables
    >(GetAuthorDocument, { id }, { fetchOptions })

    return {
      data: data ? authorFormatter(data) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)

    return { data: null, error }
  }
}
