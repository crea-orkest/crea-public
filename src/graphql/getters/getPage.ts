import { client } from "../gqlClient";
import { pageQueryFormatter } from "graphql/formatters/pageQueryFormatter";
import {
  GetPageDocument,
  type GetPageQuery,
  type GetPageQueryVariables,
} from "../generated/graphql";

export const getPage = async ({ slug }: GetPageQueryVariables) => {
  try {
    const { data, error } = await client.query<
      GetPageQuery,
      GetPageQueryVariables
    >(GetPageDocument, { slug });

    return {
      data: data ? pageQueryFormatter(data) : null,
      errors: error,
    };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);
    return { data: null, errors };
  }
};
