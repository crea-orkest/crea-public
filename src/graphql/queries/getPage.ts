import gql from "graphql-tag";
import { pageDetail } from "../fragments/pageDetail";

export const GET_PAGE = gql`
  query getPage($slug: String) {
    page(filter: { slug: { eq: $slug } }) {
      ...pageDetail
    }
  }
  ${pageDetail}
`;
