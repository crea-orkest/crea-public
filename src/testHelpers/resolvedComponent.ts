export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
