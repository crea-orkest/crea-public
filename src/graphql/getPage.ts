import { gqlClient } from "./gqlClient";
import { GET_PAGE } from "./queries/getPage";
import { pageQueryFormatter } from "./formatters/pageQueryFormatter";
import { GetPageQuery, GetPageQueryVariables } from "./generated/types";

export const getPage = async ({ slug }: GetPageQueryVariables) => {
  try {
    const { data, loading, errors } = await gqlClient.query<GetPageQuery>({
      query: GET_PAGE,
      variables: { slug },
    });

    return { data: pageQueryFormatter(data), loading, errors };
  } catch (error) {
    if (error instanceof Error) console.log(error.message);
    return { data: null, loading: false, error };
  }
};
