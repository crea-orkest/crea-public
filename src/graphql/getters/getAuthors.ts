import { GET_AUTHORS } from "graphql/queries/getAuthors";
import { authorsFormatter } from "../formatters/authorsFormatter";
import {
  GetAuthorsQuery,
  GetAuthorsQueryVariables,
} from "../generated/graphql";
import { getClient } from "../gqlClient";

export const getAuthors = async ({
  first,
  skip,
  order,
}: GetAuthorsQueryVariables) => {
  const client = getClient();

  try {
    const { data, loading, errors } = await client.query<GetAuthorsQuery>({
      query: GET_AUTHORS,
      variables: { first, skip, order },
    });
    return { data: authorsFormatter(data), loading, errors };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);

    return { data: null, loading: false, errors };
  }
};
