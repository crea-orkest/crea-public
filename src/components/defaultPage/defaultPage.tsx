import React from 'react'
import { getPage } from 'graphql/getters/getPage'
import { notFound } from 'next/navigation'
import styles from './styles.module.scss'

export interface Props {
  slug: string
}

export const DefaultPage = async ({ slug }: Props) => {
  const { data } = await getPage({ slug })
  if (!data) return notFound()

  return (
    <article className={styles.article}>
      <h1>{data.title}</h1>
      {JSON.stringify(data)}
    </article>
  )
}
