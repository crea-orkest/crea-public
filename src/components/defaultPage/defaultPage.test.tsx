import { DefaultPage } from './defaultPage'
import type { Props } from './defaultPage'
import React from 'react'
import { getPage } from 'graphql/getters/getPage'
import { resolvedComponent } from '../../testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('../../graphql/getters/getPage', () => {
  return {
    __esModule: true,
    getPage: jest.fn(),
  }
})

const getPageMock = jest.mocked(getPage)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getPageMock.mockResolvedValueOnce({
      data: {
        title: 'title',
        slug: 'contact',
        _createdAt: 'some date time string',
        _updatedAt: 'some date time string',
        id: 'unique-id',
        _seoMetaTags: [
          {
            tag: 'seo-tag',
          },
        ],
        content: [],
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(DefaultPage, {
      slug: 'contact',
    })

    render(<Resolved />)

    expect(screen.getByText('title')).toBeTruthy()
  })
})
