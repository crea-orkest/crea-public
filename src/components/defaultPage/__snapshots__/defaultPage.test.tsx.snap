// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <article
    class="article"
  >
    <h2>
      title
    </h2>
  </article>
</div>
`;
